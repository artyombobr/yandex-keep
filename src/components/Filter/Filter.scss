@import "../../shared";

.tags {
  overflow: hidden;
  height: 64px;
  margin-top: 80px;

  &__inner {
    display: flex;
    align-items: center;
    overflow-x: scroll;
    white-space: nowrap;
    padding-bottom: 30px;
    width: 100%;
  }

  &__title {
    display: none;
    font-size: 36px;
    line-height: 42px;
    font-weight: normal;
    margin-left: 16px;
  }

  & input {
    display: none;
  }
}

.tag {
  margin-left: 16px;
}

.tag__span {
  display: block;
  position: relative;
  height: 48px;
  width: 48px;
  border-radius: 8px;
  opacity: .4;
  cursor: pointer;
  border: none;
  outline: none;

  &:hover {
    box-shadow: 0 2px 2px rgba(0, 0, 0, .25);
    opacity: 1;
  }

  &::after {
    content: "";
    display: none;
    position: absolute;
    top: 15px;
    left: 10px;
    width: 17px;
    height: 8px;
    border: 4px none;
    border-left: 5px solid;
    border-bottom-style: solid;
    transform: translate(3px, 0) rotate(-45deg);
  }
}

input:checked ~ .tag__span {
  box-shadow: 0 2px 2px rgba(0, 0, 0, .25);
  opacity: 1;
}

input:checked ~ .tag__span::after {
  display: block;
}

@media (min-width: 1200px) {
  .tags {
    margin-top: 96px;

    &__title {
      display: block;
      margin-left: 173px;
    }
  }

  .tag__span {
    height: 32px;
    width: 32px;

    &::after {
      top: 10px;
      left: 5px;
      width: 11px;
      height: 6px;
      border: 3px none;
      border-left: 4px solid;
      border-bottom-style: solid;
    }
  }
}
