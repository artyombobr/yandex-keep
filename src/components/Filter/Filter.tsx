import React from 'react';
import './Filter.scss';

import { ColorsEntity } from '../../shared';

function Filter(props: { colors: ColorsEntity[] }) {
  const colors = props.colors;
  return (
    <section className='tags'>
      <div className='tags__inner'>
        <h1 className='tags__title'>Заметки</h1>
        {colors.map((color: ColorsEntity) => {
          const styles = {
            backgroundColor: `${color.color}`
          };
          return (
            <label key={color.id} className='tag'>
              <input type='checkbox' />
              <span className='tag__span' style={styles} />
            </label>
          )
        })}
      </div>
    </section>
  );
}

export default Filter;
