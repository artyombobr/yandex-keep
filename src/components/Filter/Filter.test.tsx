import React from 'react';
import { shallow } from 'enzyme';
import { Filter } from './Filter';

describe('Filter', () => {
  const colors = [
    {
      id: 0,
      color: '#3ae800',
    },
    {
      id: 1,
      color: '#0003f2',
    },
    {
      id: 2,
      color: '#00ddf2',
    },
  ];
  const allNotes = [
    {
      type: 'text',
      text: 'Купить чайник с Bluetooth\nВидел у Xiaomi на днях',
      tags: [0],
      size: 's',
      created: 1551593220000,
    },
  ];
  const dispatch = () => {};

  it('correct render', () => {
    const component = shallow(
      <Filter colors={colors} allNotes={allNotes} dispatch={dispatch} />
    );
    expect(component).toMatchSnapshot();
  });
});
