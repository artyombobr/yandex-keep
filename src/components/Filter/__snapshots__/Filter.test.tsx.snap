// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    allNotes={
      Array [
        Object {
          "created": 1551593220000,
          "size": "s",
          "tags": Array [
            0,
          ],
          "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
          "type": "text",
        },
      ]
    }
    colors={
      Array [
        Object {
          "color": "#3ae800",
          "id": 0,
        },
        Object {
          "color": "#0003f2",
          "id": 1,
        },
        Object {
          "color": "#00ddf2",
          "id": 2,
        },
      ]
    }
    dispatch={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="tags__inner"
      >
        <h1
          className="tags__title"
        >
          Заметки
        </h1>
        <label
          className="tag"
        >
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="tag__span"
            style={
              Object {
                "backgroundColor": "#3ae800",
              }
            }
          />
        </label>
        <label
          className="tag"
        >
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="tag__span"
            style={
              Object {
                "backgroundColor": "#0003f2",
              }
            }
          />
        </label>
        <label
          className="tag"
        >
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="tag__span"
            style={
              Object {
                "backgroundColor": "#00ddf2",
              }
            }
          />
        </label>
      </div>,
      "className": "tags",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="tags__title"
          >
            Заметки
          </h1>,
          Array [
            <label
              className="tag"
            >
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />
              <span
                className="tag__span"
                style={
                  Object {
                    "backgroundColor": "#3ae800",
                  }
                }
              />
            </label>,
            <label
              className="tag"
            >
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />
              <span
                className="tag__span"
                style={
                  Object {
                    "backgroundColor": "#0003f2",
                  }
                }
              />
            </label>,
            <label
              className="tag"
            >
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />
              <span
                className="tag__span"
                style={
                  Object {
                    "backgroundColor": "#00ddf2",
                  }
                }
              />
            </label>,
          ],
        ],
        "className": "tags__inner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Заметки",
            "className": "tags__title",
          },
          "ref": null,
          "rendered": "Заметки",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />,
              <span
                className="tag__span"
                style={
                  Object {
                    "backgroundColor": "#3ae800",
                  }
                }
              />,
            ],
            "className": "tag",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "tag__span",
                "style": Object {
                  "backgroundColor": "#3ae800",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />,
              <span
                className="tag__span"
                style={
                  Object {
                    "backgroundColor": "#0003f2",
                  }
                }
              />,
            ],
            "className": "tag",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "tag__span",
                "style": Object {
                  "backgroundColor": "#0003f2",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />,
              <span
                className="tag__span"
                style={
                  Object {
                    "backgroundColor": "#00ddf2",
                  }
                }
              />,
            ],
            "className": "tag",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "tag__span",
                "style": Object {
                  "backgroundColor": "#00ddf2",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "label",
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="tags__inner"
        >
          <h1
            className="tags__title"
          >
            Заметки
          </h1>
          <label
            className="tag"
          >
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="tag__span"
              style={
                Object {
                  "backgroundColor": "#3ae800",
                }
              }
            />
          </label>
          <label
            className="tag"
          >
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="tag__span"
              style={
                Object {
                  "backgroundColor": "#0003f2",
                }
              }
            />
          </label>
          <label
            className="tag"
          >
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="tag__span"
              style={
                Object {
                  "backgroundColor": "#00ddf2",
                }
              }
            />
          </label>
        </div>,
        "className": "tags",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="tags__title"
            >
              Заметки
            </h1>,
            Array [
              <label
                className="tag"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="tag__span"
                  style={
                    Object {
                      "backgroundColor": "#3ae800",
                    }
                  }
                />
              </label>,
              <label
                className="tag"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="tag__span"
                  style={
                    Object {
                      "backgroundColor": "#0003f2",
                    }
                  }
                />
              </label>,
              <label
                className="tag"
              >
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="tag__span"
                  style={
                    Object {
                      "backgroundColor": "#00ddf2",
                    }
                  }
                />
              </label>,
            ],
          ],
          "className": "tags__inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Заметки",
              "className": "tags__title",
            },
            "ref": null,
            "rendered": "Заметки",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />,
                <span
                  className="tag__span"
                  style={
                    Object {
                      "backgroundColor": "#3ae800",
                    }
                  }
                />,
              ],
              "className": "tag",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "tag__span",
                  "style": Object {
                    "backgroundColor": "#3ae800",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />,
                <span
                  className="tag__span"
                  style={
                    Object {
                      "backgroundColor": "#0003f2",
                    }
                  }
                />,
              ],
              "className": "tag",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "tag__span",
                  "style": Object {
                    "backgroundColor": "#0003f2",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />,
                <span
                  className="tag__span"
                  style={
                    Object {
                      "backgroundColor": "#00ddf2",
                    }
                  }
                />,
              ],
              "className": "tag",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "tag__span",
                  "style": Object {
                    "backgroundColor": "#00ddf2",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
